include/alpaka/event/EventGenericThreads.hpp

```diff
@@ -294,14 +295,16 @@ namespace alpaka
 
                 if(!spEventImpl->isReady())
                 {
-                    auto const enqueueCount = spEventImpl->m_enqueueCount;
+                    auto oldFuture = spEventImpl->m_future;
 
                     // Enqueue a task that waits for the given event.
                     queueImpl.m_workerThread.submit(
-                        [spEventImpl, enqueueCount]() mutable
+                        [spEventImpl, oldFuture]() mutable
                         {
                             std::unique_lock<std::mutex> lk2(spEventImpl->m_mutex);
-                            spEventImpl->wait(enqueueCount, lk2);
+                            lk2.unlock();
+                            oldFuture.get();
+                            lk2.lock();
```

Comment:
```suggestion
                            lk2.unlock();
                            oldFuture.get();
```
Why would the last lock be necessary? It is immediately unlocked again at the closing brace.

Reply:
You are right it is an left over from a version I used before where I was required to change a member of the event